pandas/core/arrays/categorical.py

```diff
@@ -2001,14 +2001,14 @@ def __getitem__(self, key):
             else:
                 return self.categories[i]
 
-        if is_list_like(key) and not is_array_like(key):
-            key = np.asarray(key)
-
-        if com.is_bool_indexer(key):
-            key = check_bool_array_indexer(self, key)
+        if is_list_like(key) and not isinstance(key, tuple):
+            key = check_array_indexer(self, key)
 
         result = self._codes[key]
         if result.ndim > 1:
+            from pandas.core.indexes.base import deprecate_ndim_indexing
```

Comment:
can this be too imported?

Reply:
It seems this is possible yes. But I don't really like array code importing code from the Index classes (the dependence should be the other way around). I can maybe also move the `deprecate_ndim_indexing` helper function to `pd.core.indexers` (instead of `pd.core.indexes`) to have this separation cleaner.

Reply:
+1

Reply:
hmm yeah I agree with this as its not specific to index. in factor I would move to pandas.compat.numpy_